### ISSUE TITLE
How to generate the .labels for semantic-kitti test dataset for online evaluation?

### ISSUE THREAD

**Issue by densechen** [HuguesTHOMAS/KPConv-PyTorch#75]
Hi, @HuguesTHOMAS , thanks for providing such nice work.

How can we generate the .labels file used to online evaluation?

Best

**Comment by HuguesTHOMAS** [comment]
Oh, it is true I did not write the code for that.

You can refer to the original TensorFlow implementation that I provided before [here](https://drive.google.com/open?id=12npkHHnqzhhl5i-2q_RD-Cw_urUdWC0J).

In there you will find a file called `create_SemanticKitti_submission.py`. You will probably have to adapt it for it to work with this version of the code but the essential is there.

**Comment by densechen** [comment]
Thanks a lot, and I will try with it.

**Comment by songw-zju** [comment]
I use the file `create_SemanticKitti_submission.py` to get .labels for seq08 and evaluate the results with .labels. But the miou is lower than results during training process. Have you encountered this problem？Thanks in advance.

**Comment by xuanyi97** [comment]
Hi, could you share the code to generate labels for a sequence folder for semantickitti dataset?

**Comment by magical1998** [comment]
> 我使用该文件`create_SemanticKitti_submission.py`获取了 seq08 的 .labels，并用 .labels 评估结果。但是 miou 明显低于训练过程中的结果。你遇到过这个问题吗？先谢谢了。
Hello, this is what happened to the graph I generated, how did you fix it then?
![Uploading image.png…]()

**Comment by magical1998** [comment]
![image](https://github.com/user-attachments/assets/f77b33b5-ee67-4f5e-91a0-c0f7065108dd)
